corehq/ex-submodules/phonelog/models.py

```diff
@@ -24,6 +24,39 @@ class Meta:
             ("domain", "date"),
         ]
 
+    def __repr__(self):
+        return u"DeviceReportEntry(domain='{}', msg='{}')".format(self.domain, self.msg)
+
+
+class UserErrorEntry(models.Model):
+    # Information about the device log this came from
+    domain = models.CharField(max_length=100, db_index=True)
+    xform_id = models.CharField(max_length=COUCH_UUID_MAX_LEN, db_index=True)
+    i = models.IntegerField()
+
+    # The context around when/how this happened
+    app_id = models.CharField(max_length=COUCH_UUID_MAX_LEN)
+    version_number = models.IntegerField()
+    date = models.DateTimeField()
+    server_date = models.DateTimeField(null=True, db_index=True)
+    user_id = models.CharField(max_length=COUCH_UUID_MAX_LEN, db_index=True)
+
+    # Information about the specific error
+    expr = models.TextField()
+    msg = models.TextField()
+    session = models.TextField()
+    type = models.CharField(max_length=32, db_index=True)
+
+    class Meta:
+        app_label = 'phonelog'
+        unique_together = [('xform_id', 'i')]
+        index_together = [
+            ("app_id", "version_number"),
+        ]
```

Comment:
how will you be querying for these? i don't think this is a lot. we may want to include domain since that's almost always in the query